### PR TITLE
Add main classes

### PR DESCRIPTION
**Pull request by m-talbi**
- Added Person class
- Added Student class derived from Person class.
- Added Teacher class derived from Person class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Person
+  attr_reader :parent_permission
```
Comment (Munsa1) [inline person.rb]: - [ ] `id` should be able to be retrieved so the Getter for `id` should be defined. Kindly replace `:parent_permission` with `id`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
```
Comment (Munsa1) [inline person.rb]: - [ ] Kindly initialize `id` here as well, and the `id` should be randomly generated using the random function as this is a requirement.
e.g `@id = Random.rand(1..1000)`

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require_relative 'person'
+class Student < Person
+  def initialize(age, classroom, name, parent_permission)
```
Comment (Munsa1) [inline student.rb]: - [ ] the `name` attribute should be set to `"Unknown"` and the `parent_permission` should be set to `true`

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require_relative 'person'
+class Teacher < Person
+  def initialize(specialization, age, name, parent_permission)
```
Comment (Munsa1) [inline teacher.rb]: - [ ] again here the `name` attribute should be set to `"Unknown"` and the `parent_permission` should be set to `true`

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Kweeka1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = rand(1..1000)
```
Comment (Reem-lab) [inline person.rb]: - [ OPTIONAL ] I think rand comes from a Class called Random so please add it before rand method to make random id works as expected ✨♻🔥💯

**Review by Reem-lab — APPROVED** [review]
# Hi @Kweeka1, 👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- Followed Gitflow ✅
- Great working on the functionality ✅
-  Professional README ✅
- Descriptive PR 💯💯


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**